topdown/cidr.go

```diff
@@ -0,0 +1,35 @@
+package topdown
+
+import (
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/topdown/builtins"
+	"net"
+)
+
+func builtinNetCIDROverlap(a, b ast.Value) (ast.Value, error) {
+	pattern, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	match, err := builtins.StringOperand(b, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	_, cidrnet, err := net.ParseCIDR(string(pattern))
+	if err != nil {
+		return nil, err
+	}
+
+	ip := net.ParseIP(string(match))
+	if ip == nil {
+		return nil, err
```

Comment:
What is the `err` here ? We can return something like `fmt.Errorf("Invalid IP representation")`

Reply:
Yeah, totally missed that - thanks!